packages/framework/tests/Unit/Component/Doctrine/QueryBuilderExtenderTest.php

```diff
@@ -91,4 +92,35 @@ public function extendJoinWithExtendedEntityProvider(): array
             ],
         ];
     }
+
+    public function testDifferentAlias()
+    {
+        /** @var \Doctrine\ORM\EntityManager $entityManager */
+        $entityManager = $this->getMockBuilder(EntityManagerInterface::class)
+            ->disableOriginalConstructor()
+            ->getMockForAbstractClass();
+
+        $queryBuilder = new QueryBuilder($entityManager);
+        $queryBuilder
+            ->select('c')
+            ->from(Category::class, 'c')
+            ->join(Product::class, 'p', Join::WITH, 'p.id = 1');
+
+        $entityNameResolver = new EntityNameResolver([]);
+        $queryBuilderExtender = new QueryBuilderExtender($entityNameResolver);
+        $queryBuilderExtender->addOrExtendJoin(
+            $queryBuilder,
+            Product::class,
+            'p2',
+            'p2.id = 2'
+        );
+
+        $dql = $queryBuilder->getDQL();
+        $this->assertSame(
+            'SELECT c FROM ' . Category::class . ' c'
+            . ' INNER JOIN ' . Product::class . ' p WITH p.id = 1'
+            . ' INNER JOIN ' . Product::class . ' p2 WITH p2.id = 2',
+            $dql
+        );
+    }
```

Comment:
😍

Reply:
Yeah! Thanks to @pk16011990 for excelent issue 👏